nuxeo-features/nuxeo-admin-center/nuxeo-connect-client-wrapper/src/main/java/org/nuxeo/connect/client/ConnectClientComponent.java

```diff
@@ -69,7 +72,11 @@ public void activate(ComponentContext context) {
         }
 
         if (adapter.getCanonicalName().equals(PackageManager.class.getCanonicalName())) {
-            return adapter.cast(NuxeoConnectClient.getPackageManager());
+            String distribName = Framework.getProperty(Environment.DISTRIBUTION_NAME);
+            String distribVersion = Framework.getProperty(Environment.DISTRIBUTION_VERSION);
+            String targetPlatform = (distribName != null && distribVersion != null) ? distribName + "-" + distribVersion
+                    : null;
+            return adapter.cast(NuxeoConnectClient.getPackageManager(targetPlatform));
         }
```

Comment:
Can you please explain what will be the behaviour here on different use cases than a release? (e.g SNAPSHOT distribution, datebased distribution, and HF distribution -- this one would never be used if users use the mp-hotfix command on a release, but handling it would still be useful for some edge cases)

Reply:
added some tests in the next commits